components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceEventManagementServiceImpl.java

```diff
@@ -363,8 +363,13 @@ public Response getLastKnownData(@PathParam("deviceId") String deviceId, @PathPa
             List<SortByField> sortByFields = new ArrayList<>();
             SortByField sortByField = new SortByField(TIMESTAMP_FIELD_NAME, SortType.DESC);
             sortByFields.add(sortByField);
+            if(limit == 0){
             EventRecords eventRecords = getAllEventsForDevice(sensorTableName, query, sortByFields, 0, 1);
-            return Response.status(Response.Status.OK.getStatusCode()).entity(eventRecords).build();
+            return Response.status(Response.Status.OK.getStatusCode()).entity(eventRecords).build();}
+            else{
```

Comment:
@lashanfaliq95 There is a formatting issue here.  Please apply the code formatting to all the files you have changed.